### DISCUSSION TITLE
Is a linear transform on an Image supposed to be applied in place?

### DISCUSSION THREAD

**Discussion by baba-yaga** [marcomusy/vedo#1011] [Q&A] [answered]
It seems Image.clone() does not pick up applied transforms, or these are not applied in place, as the documentation suggests. Consider an example:

```
from vedo import *
img = Image(dataurl+"images/dog.jpg")  # 3D embedding of the image

plt = Plotter(axes=1)
plt += img.alpha(0.5)
img1 = img.clone().scale(0.5)
plt += img1.shift((-100, -100, 100))
plt.show().close()  # result as expected

```
The result as expected:
<img width="384" alt="Screenshot 2024-01-09 at 16 08 28" src="https://github.com/marcomusy/vedo/assets/2671379/4a39331a-984b-493b-9f71-b66a20c6fbd3">

Now, use img1.clone() instead:

```
plt = Plotter(axes=1)
plt += img.alpha(0.5)
img1 = img.clone().scale(0.5)
plt += img1.clone().shift((-100, -100, 100))
plt.show().close()  # scaling is lost!

```
<img width="380" alt="Screenshot 2024-01-09 at 16 09 50" src="https://github.com/marcomusy/vedo/assets/2671379/8dc2b1b7-3b66-47a3-95fe-9a70b1bed48a">

The scaling is lost! It look that it is img but not img1 is cloned

**Comment by marcomusy** [comment, marked as answer]
Thanks - I just fixed it in the master.

```py
from vedo import *

img = Image(dataurl+"images/dog.jpg").alpha(0.75)
img1 = img.clone()
img1.scale(0.25)
img1.shift(-100, -100, 100)
show(img, img1, axes=1).close() 

img = Image(dataurl+"images/dog.jpg").alpha(0.75)
img1 = img.clone()
img1.scale(0.25)
img2 = img1.clone()
img2.shift(-100, -100, 100)
show(img, img2, axes=1).close() 
```

![Screenshot from 2024-01-10 15-43-38](https://github.com/marcomusy/vedo/assets/32848391/efee7f86-7949-4735-8394-6ad71af61366)
